src/VASSAL/build/module/Chatter.java

```diff
@@ -21,477 +21,636 @@
+        } catch (BadLocationException ble) {
+          ErrorDialog.bug(ble);
+        } catch (IOException ex) {
+          ErrorDialog.bug(ex);
+      }
     }
-    conversation.setEditable(false);
-    conversation.setLineWrap(true);
-    conversation.setWrapStyleWord(true);
-    conversation.setUI(new UI());
-    conversation.addComponentListener(new ComponentAdapter() {
-      @Override
+
+		conversation.setEditable(false);
+		conversation.addComponentListener(new ComponentAdapter() {
+			@Override
       public void componentResized(ComponentEvent e) {
-        scroll.getVerticalScrollBar().setValue(scroll.getVerticalScrollBar().getMaximum());
-      }
-    });
-    input = new JTextField(80);
-    input.setFocusTraversalKeysEnabled(false);
-    input.addActionListener(new ActionListener() {
-      @Override
+				scroll.getVerticalScrollBar().setValue(scroll.getVerticalScrollBar().getMaximum());
+			}
+		});
+
+		input = new JTextField(60);
+		input.setFocusTraversalKeysEnabled(false);
+		input.addActionListener(new ActionListener() {
+			@Override
       public void actionPerformed(ActionEvent e) {
-        send(formatChat(e.getActionCommand()));
-        input.setText(""); //$NON-NLS-1$
-      }
-    });
-    input.setMaximumSize(new Dimension(input.getMaximumSize().width,
-                                       input.getPreferredSize().height));
-    scroll.setViewportView(conversation);
-    add(scroll);
-    add(input);
-  }
+				send(formatChat(e.getActionCommand()));
+				input.setText(""); //$NON-NLS-1$
+			}
+		});
+		input.setMaximumSize(new Dimension(input.getMaximumSize().width, input.getPreferredSize().height));
 
-  private String formatChat(String text) {
-    final String id = GlobalOptions.getInstance().getPlayerId();
-    return "<" + (id.length() == 0 ? "("+getAnonymousUserName()+")" : id) + "> - " + text; //$NON-NLS-1$ //$NON-NLS-2$
-  }
+		conversation.setPreferredSize(new Dimension(input.getMaximumSize().width, input.getPreferredSize().height * 11));
 
-  public JTextField getInputField() {
-    return input;
-  }
-
-  /**
-   * Display a message in the text area
-   */
-  public void show(String s) {
-    conversation.append("\n" + s); //$NON-NLS-1$
-  }
-
-  /** @deprecated use GlobalOptions.getPlayerId() */
-  @Deprecated public void setHandle(String s) {
-  }
+		scroll.setViewportView(conversation);
+		scroll.getVerticalScrollBar().setUnitIncrement(input.getPreferredSize().height); // BR// Scroll this faster																									
+		add(scroll);
+		add(input);
+	}
 
-  /** @deprecated use GlobalOptions.getPlayerId() */
-  @Deprecated public String getHandle() {
-    return GlobalOptions.getInstance().getPlayerId();
-  }
+	private String formatChat(String text) {
+    final String id = GlobalOptions.getInstance().getPlayerId();
+    return "&lt;" + (id.length() == 0 ? "(" + getAnonymousUserName() + ")" : id) + "&gt; - " + text; //$NON-NLS-1$ //$NON-NLS-2$ //BR//HTML-friendly angle brackets
```

Comment:
`String.isEmpty()` is a slightly better readable way to express `String.length() == 0`.

Also, String.format() is a bit more readable in these cases, e.g.:
`
return String.format("&lt;%s&gt; - %s", id.length() == 0 ? "(" + getAnonymousUserName() + ")" : id, text);
`

or even:
`
final String idPart = id.isEmpty() ? "(" + getAnonymousUserName() + ")" : id;
return String.format("&lt;%s&gt; - %s", idPart, text);
`

Reply:
fixed for next push